### PR TITLE
Patch to debian12

### PR DESCRIPTION
**Pull request by AaronMoat**
Resolves https://github.com/seek-oss/skuba/issues/1380

### CODE REVIEW

File: src/cli/configure/upgrade/patches/7.3.1/patchDockerfile.ts [hunk 1]
```diff
@@ -8,8 +8,9 @@ import { createDestinationFileReader } from '../../../analysis/project';
 
 const DOCKERFILE_FILENAME = 'Dockerfile';
 
-const VERSION_REGEX = /gcr.io\/distroless\/nodejs:(16|18|20)/g;
-const VERSION_DEBIAN_REPLACE = 'gcr.io/distroless/nodejs$1-debian11';
+const NON_DEBIAN_REGEX = /gcr.io\/distroless\/nodejs:(16|18|20)/g;
+const DEBIAN_REGEX = /gcr.io\/distroless\/nodejs(16|18|20)-debian11/g;
```
Comment (72636c) [inline src/cli/configure/upgrade/patches/7.3.1/patchDockerfile.ts:12]: It looks like there is no `nodejs16-debian12` support: https://console.cloud.google.com/gcr/images/distroless

Comment (AaronMoat) [reply src/cli/configure/upgrade/patches/7.3.1/patchDockerfile.ts:12]: Ooh that's complicated then. Maybe it should be two patches

Comment (AaronMoat) [reply src/cli/configure/upgrade/patches/7.3.1/patchDockerfile.ts:12]: Ah, but skuba has engineConfig requiring 18 so this won't run on 16 anyway - maybe we just do the 18/20

Comment (72636c) [reply src/cli/configure/upgrade/patches/7.3.1/patchDockerfile.ts:12]: Oh nice one, sounds good 👍

**Review by 72636c — APPROVED** [review]

**Pull request merged**